src/main/webapp/ratingScript.js

```diff
@@ -0,0 +1,128 @@
+  url.searchParams.set("term-key", ratingProperties.termKey);
+  url.searchParams.set("hour", ratingProperties.hours);
+  url.searchParams.set("difficulty", ratingProperties.difficulty);
+  url.searchParams.set("term-input", ratingProperties.termInput);
+  url.searchParams.set("prof-input", ratingProperties.profInput);
+  url.searchParams.set("rating-term", ratingProperties.ratingTerm);
+  url.searchParams.set("rating-prof", ratingProperties.ratingProf);
+
+  return url;
+}
+
+async function passRatingProperties() {
+  document.getElementById("message-rating").innerHTML = "";
+  const urlAndData = await getRatingPropertiesToStore();
+  postRatingProperties(urlAndData[0], urlAndData[1]);
+}
+
+$(function () {
+  $('[data-toggle="tooltip"]').tooltip();
+});
+
+async function getLatestRating() {
+  document
+    .getElementById("message-rating")
+    .classList.remove("alert-light");
+  document
+    .getElementById("message-rating")
+    .classList.add("alert-secondary");
+  document.getElementById("message-rating").innerHTML =
+    "Fetching your rating...";
+
+  const userId = await verify();
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  const termKey = urlParams.get("term-key");
+
+  const url = new URL("/latest-rating", window.location.origin);
+  url.searchParams.set("reviewer-id", userId);
+  url.searchParams.set("term-key", termKey);
+  const response = await fetch(url);
+  const formInfo = await response.json();
+
+  if (Object.keys(formInfo).length == 0) {
+    document
+      .getElementById("message-rating")
+      .classList.remove("alert-secondary");
+    document
+      .getElementById("message-rating")
+      .classList.add("alert-danger");
+    document.getElementById("message-rating").innerHTML =
+      "You have not submitted a rating for this term";
+  } else {
+    document.getElementById("term-input").innerHTML = formInfo["termInput"];
+    document.getElementById("prof-input").innerHTML = formInfo["profInput"];
+    document.getElementById("rating-term").value = formInfo["ratingTerm"];
+    document.getElementById("rating-prof").value = formInfo["ratingProf"];
+    document.getElementById("hours").value = formInfo["hours"];
+    document.getElementById("difficulty").value = formInfo["difficulty"];
+    document.getElementById("grade").value = formInfo["grade"];
+    document.getElementById("translate").value = formInfo["translation"];
+
+    document.getElementById("message-rating").innerHTML =
+      "Your form has been populated!";
+    document
+      .getElementById("message-rating")
+      .classList.remove("alert-secondary");
+    document
+      .getElementById("message-rating")
+      .classList.add("alert-success");
+  }
+}
+
+document
+  .getElementById("form-submit")
+  .addEventListener("click", passRatingProperties);
+
+document
+  .getElementById("latest-rating")
+  .addEventListener("click", getLatestRating);
```

Comment:
These are happening globally? You could just add this to the html directly, onClick="passRatingProperties()" etc I think. 
Though if this works I guess it's fine.

Reply:
For some reason that was not working for us before, but I will try to add it back and see if that got fixed!